gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java

```diff
@@ -86,8 +89,8 @@ protected T computeNext() {
     if (!entityResultPbIter.hasNext()) {
       return endOfData();
     }
-    DatastoreV1.EntityResult entityResultPb = entityResultPbIter.next();
-    //cursor = entityResultPb.getCursor(); // only available in v1beta3
+    com.google.datastore.v1beta3.EntityResult entityResultPb = entityResultPbIter.next();
+    cursor = entityResultPb.getCursor();
```

Comment:
Sorry I don't understand. Each cursor in the EntityResult corresponds to the point right after the accompanying entity. Additionally, the end cursor points to the point right after the end of the batch. Usually the end cursor would be the same as the last entity cursor in the batch. However, if the Datastore makes progress after the last result the end cursor could be the same as the last result cursor.